aztec/src/main/kotlin/org/wordpress/aztec/formatting/BlockFormatter.kt

```diff
@@ -212,6 +214,80 @@ class BlockFormatter(editor: AztecText,
         return changed
     }
 
+    /**
+     * This method makes sure only one block style is ever applied to part of the text. The following block styles are
+     * made exclusive if the option is enabled:
+     * - all the lists
+     * - all the headings
+     * - quote
+     * - preformat
+     */
+    private fun removeBlockStylesFromSelectedLine(appliedClass: IAztecBlockSpan) {
+        // We only want to remove the previous block styles if this option is enabled
+        if (!exclusiveBlockStyles.enabled) {
+            return
+        }
+        val selectionStart = editor.selectionStart
+
+        // try to remove block styling when pressing backspace at the beginning of the text
+        editableText.getSpans(selectionStart, selectionEnd, IAztecBlockSpan::class.java).forEach {
+            // We want to remove any list item span that's being converted to another block
+            if (it is AztecListItemSpan) {
+                editableText.removeSpan(it)
+                return@forEach
+            }
+            // Only these supported blocks will be split/removed on block change
+            val format = when (it) {
+                is AztecHeadingSpan -> it.textFormat
+                is AztecOrderedListSpan -> AztecTextFormat.FORMAT_ORDERED_LIST
+                is AztecUnorderedListSpan -> AztecTextFormat.FORMAT_UNORDERED_LIST
+                is AztecTaskListSpan -> AztecTextFormat.FORMAT_TASK_LIST
+                is AztecListItemSpan -> AztecTextFormat.FORMAT_UNORDERED_LIST
+                is AztecQuoteSpan -> AztecTextFormat.FORMAT_QUOTE
+                is AztecPreformatSpan -> AztecTextFormat.FORMAT_PREFORMAT
+                else -> return@forEach
```

Comment:
This is the only concern I have about the PR. It seems that if I extend Aztec with a block span and I wanted to be exclusive, I have to come to this place and add it as well. This then becomes an [unknown unknown](https://medium.com/@nathan.fooo/notes-a-philosophy-of-software-design-f48206bb32de) because I need to know about this piece of code to make it work. 

Maybe if the `Span` has the information about whether or not the block is exclusive and also holds the `textFormat`. With this design, then it becomes obvious to the programmer who is going to add a new block and want it to be exclusive. 

WDYT?

Reply:
that's a good point 👍. I agree and I'll try to improve it